Clarifications (#5)
w3c-fedid/Administration: proposals-CG-WG.md

# CG/WG Proposal Stages

Author: @samuelgoto
Created: 08/11/2024
Last update: 09/11/2024
Status: under review

This is a proposal to break proposals into 5 stages of maturity, with clear guidelines and requirements to advance them:

* [Stage 0](#stage0): **Exploration** of the Problem Space
* [Stage 1](#stage1): **Incubation** of competing Alternatives
* [Stage 2](#stage2): **Formalization** a preferred Proposal
* [Stage 3](#stage3): **Independent Implementations** of the preferred Proposal
* [Stage 4](#stage4): **Publication** of a Proposed Recommendation

> 

Reviewer: Suggest adding something like:
"These stages support the W3C process. If there is ever any question between W3C process requirements and how the groups will progress their work, the W3C process has precedent."

Author: sgtm done

# Stage 0: Exploration

The purpose of Stage 0 proposals is to allow anyone to raise and explore the Problem Space without asking for permission. 

>   * What's needed from individuals?

Author: Ah, that occurred to me too, but I figured it would be constructive to separate what the champions need to bring and what the reviewers need to agree to. That seemed like an important distinction, wdyt?

>   * What's needed from individuals?

Author: done

    * Typically, a written issue, a personal repo, a blog post, a tweet, etc. 
  * What's asked of the **Community Group**?
    * Nothing: no permission needed from anyone.

# Stage 1: Incubation

> The purpose of Stage 1 proposals is to explore the Solution Space and pick a preferred direction. This is where most of the work gets done because it involves exploring alternatives, understanding tradeoffs, gathering implementation experience, incubating alternatives, gathering evidence of demand and fitness for purpose and finally identifying the best out of the many alternatives.

Reviewer 2: ```suggestion
The purpose of Stage 1 proposals is to explore the Solution Space and pick a preferred direction. This is where most of the work gets done because it involves exploring alternatives, understanding tradeoffs, gathering implementation experience, incubating alternatives, gathering evidence of demand and fitness for purpose, and finally, identifying the best out of the many alternatives.
```

  * What's needed from individuals?

>     * Identification of [champions](https://github.com/tc39/how-we-work/blob/main/champion.md)

Author: Yeah, I was planning to added text about what I mean by "champions" in this specific case, in a later PR.

    * A good understanding of the Problem Space
    * Optionally, Alternatives under consideration
  * What's asked of the **Community Group**?

>     * The Communit Group consensus that the problem is worth spending the Community Group’s time working on

Reviewer: ```suggestion
    * The Community Group consensus that the problem is worth spending the Community Group’s time working on
```

>   * What happens if accepted?

Author: Let me try to clarify that in a separate PR.

>     * Optionally, a repo gets created for the champions to develop the feature (e.g., [example](https://github.com/fedidcg/CrossSiteCookieAccessCredential)). Small features can also continue as issues.

Reviewer: ```suggestion
    * Small features will incubate in issues. If and when the champion is ready to more thoroughly document their proposal, the WG chairs will create a repo for the champions to develop the feature (e.g., [example](https://github.com/fedidcg/CrossSiteCookieAccessCredential)). 
```

>     * Optionally, a repo gets created for the champions to develop the feature (e.g., [example](https://github.com/fedidcg/CrossSiteCookieAccessCredential)). Small features can also continue as issues.

Author: done

>     * Optionally, a repo gets created for the champions to develop the feature (e.g. [example](https://github.com/fedidcg/CrossSiteCookieAccessCredential)). Small features can also continue as issues.

Reviewer 2: ```suggestion
    * Optionally, a repo gets created for the champions to develop the feature (e.g., [example](https://github.com/fedidcg/CrossSiteCookieAccessCredential)). Small features can also continue as issues.
```

# Stage 2: Formalization

> The purpose of Stage 2 Proposals is to formally specify the best (and seemingly workable) alternative that was identified in the prior step: handle corner cases, integrate with other parts, reconcile with other proposals and resolve the concerns identified at the entrance. The Proposal enters Stage 2 with a list of blocking issues to advance to the next stage and exits with all of the issues resolved.

Reviewer 2: ```suggestion
The purpose of Stage 2 Proposals is to formally specify the best (and seemingly workable) alternative that was identified in the prior step: handle corner cases, integrate with other parts, reconcile with other proposals, and resolve the concerns identified at the entrance. The Proposal enters Stage 2 with a list of blocking issues to advance to the next stage and exits with all of the issues resolved.
```

  * What's needed from champions?
    * An [explainer](https://tag.w3.org/explainers/)
    * Alternatives and trade-offs considered

>     * A specific proposal with varying degrees of specificity required for the Working Group review, e.g., code samples, examples, or a spec PRs (if the formalism is needed to understand the proposal)

Reviewer: The more I look at it, the more I think that "varying degrees of specificity" might be overspecifying the output.

Author: "A specific proposal with varying degrees of specificity required for the Working Group review, e.g., code samples, examples, or a spec PRs (if the formalism is needed to understand the proposal)"

Does this work?

Author: Resolved offline: @hlflanagan was looking at an old snapshot.

>     * A specific proposal with varying degrees of specificity required for the Working Group review, e.g. code samples, examples or a spec PRs (if the formalist is needed to understand the proposal)
>     * Browser and developer implementation experience (e.g. a prototype, dev trials, origin trials, etc) 

Reviewer 2: ```suggestion
    * A specific proposal with varying degrees of specificity required for the Working Group review, e.g., code samples, examples, or spec PRs (if the formalism is needed to understand the proposal)
    * Browser and developer implementation experience (e.g., a prototype, dev trials, origin trials, etc.) 
```

    * Evidence of developer demand and fitness for purpose
  * What's asked of the **Working Group**?
    * Working Group consensus to adopt the proposal as the basis for their work in the [Editor’s Draft](https://w3c-fedid.github.io/FedCM/).
    * The Working Group identifies a list of (seemingly resolvable) issues that have to be addressed before [Stage 3](#stage-3).

>   

Reviewer 3: Missing what happens during this stage 2 (not sure if necessary?)

Author: not necessary, i think, because nothing gets produced by the working group, i think.

Author: Ok, just following up on this discussion, I kicked off a separate PR to try to improve the "What happens if accepted" section: https://github.com/w3c-fedid/Administration/pull/7

Lets carry on there.

>     * The Working Group consensus to adopt the proposal as the basis for their work in the [Editor’s Draft](https://w3c-fedid.github.io/FedCM/).

Reviewer 2: ```suggestion
    * Working Group consensus to adopt the proposal as the basis for their work in the [Editor’s Draft](https://w3c-fedid.github.io/FedCM/).
```

# Stage 3: Independent Implementations

>  The purpose of Stage 3 Proposals is to increase the implementation and deployment confidence in order to produce a [Candidate Recommendation](https://www.w3.org/policies/process/#RecsCR).

Reviewer 2: ```suggestion
 The purpose of Stage 3 Proposals is to increase implementation and deployment confidence in order to produce a [Candidate Recommendation](https://www.w3.org/policies/process/#RecsCR).
```

 
  * What's needed from champions?
    * Proposal fully merged into the [Editor’s Draft](https://w3c-fedid.github.io/FedCM/)
    * Web Platform Tests
    * Further implementation experience
  * What's asked of the **Working Group**? 

>     * The Working Group consensus that the [Editor’s Draft](https://www.w3.org/policies/process/#editors-draft) sufficiently resolves all of the issues raised at [Stage 2](#stage-2)

Reviewer 2: ```suggestion
    * Working Group consensus that the [Editor’s Draft](https://www.w3.org/policies/process/#editors-draft) sufficiently resolves all of the issues raised at [Stage 2](#stage-2)
```

  * What happens if accepted?
    * The [Editor’s Draft](https://w3c-fedid.github.io/FedCM/) gets published as the Working Group's [Working Draft](https://www.w3.org/TR/fedcm/)
  
# Stage 4: Publication

 The purpose of Stage 4 Proposals is to produce a [W3C Recommendation](https://www.w3.org/policies/process/#RecsW3C).
 
  * What's needed from champions?

>     * Two independent and interoperable implementations

Reviewer 3: This will be a critical requirement so it is important to be precise. Do two Chromium browsers count? Or does it need to be implemented by two independent rendering engines?

Author: Good question, but will leave that clarification to a separate PR with other reviewers too. It is ultimately the bar that the W3C sets, I think.

Author: Ok, resolving this in this PR:

https://github.com/w3c-fedid/Administration/pull/9

Resolving this here.

    * Significant deployment experience
  * What's asked of the **Working Group**?

>     * The Working Group consensus that the [Working Draft](https://www.w3.org/policies/process/#RecsWD) as a whole is complete and can be published as a [Candidate Recommendation](https://www.w3.org/policies/process/#RecsCR)

Reviewer 2: ```suggestion
    * Working Group consensus that the [Working Draft](https://www.w3.org/policies/process/#RecsWD) as a whole is complete and can be published as a [Candidate Recommendation](https://www.w3.org/policies/process/#RecsCR)
```
